README.md

```diff
@@ -1,19 +1,153 @@
-![CI](https://github.com/Cardinal-Cryptography/rush/workflows/CI/badge.svg)
+[![Crate][crate-image]][crate-link]
+[![Docs][docs-image]][docs-link]
+[![Build Status][build-image]][build-link]
+[![Apache 2.0 Licensed][license-image]][license-link]
+![Rust Stable][rustc-image]
 
-# rush
+### Aleph Consensus
 
-rust implementation of Aleph protocol
+![aleph logo](logo.jpg "Aleph logo")
 
-# examples
+### Overview
 
-There is a dummy implementation of an honest committee member that is not
+Aleph is an asynchronous and Byzantine fault tolerant consensus protocol aimed
+at ordering arbitrary messages (transactions). It has been designed to operate
+continuously under conditions where there is no bound on message-delivery delay
+and under the assumption that there is a significant probability of malicious
+behavior, making it an excellent fit for blockchain-related applications.
+For more information, check [the paper][paper-link]
+
+This repository contains a Rust implementation of Aleph that offers a convenient
+API enabling seamless application to various problems. The prime application of
+the repository is the consensus engine (sometimes called the "finality gadget")
+of the [Aleph Zero blockchain][aleph-node-link].
+
+### Detailed documentation
+
+If the crate's [documentation][docs-link] seems to be not comprehensive enough,
+please refer to the [detailed version][reference-link].
+
+### Implementation status
+
+- current version is asynchronous, so it's driven by consensus events as opposed
+  to some clock ticks
+- while being asynchronous, the performance is still optimal in partially
+  synchronous environment
+- guaranteed safety in partially synchronous environment
+- BFT - secure if less than one third of the committee is malicious
+- secure against fork bombs, for details see [the paper][paper-link]
+- network overhead optimized to not send all parents hashes but a bitmap and a control hash
+- thorough testing, including malicious scenarios, and high code coverage
+
+### Future work
+
+- asynchronous liveness - current version is not secure in a rather theoretic
+  scenario when an adversary controls all the delays in the network, including
+  honest nodes messages. However, we take security very seriously, so we plan to
+  make the consensus secure even in such a unusual scenario.
```

Comment:
asynchronous liveness - the current version is not live under full asynchrony. It is unclear whether this has any practical relevance, but we are making efforts to strenghen the guarantees in this direction. Adding a safe source of randomness to the protocol allows to make it fully asynchronous.

Reply:
Changed.